pkg/resource/deploy/step_generator.go

```diff
@@ -1209,6 +1252,9 @@ func (sg *stepGenerator) generateStepsFromDiff(
 }
 
 func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets) ([]Step, error) {
+	sg.urnLock.lock()
+	defer sg.urnLock.unlock()
```

Comment:
does this need to be under the lock? this work is never done concurrently with other step generation, right? or has that changed with this PR?

Reply:
This doesn't _need_ to be under the lock, I just took the principle of ensuring that every public method of the step generator was synchronized. I believe that for deletes this will be inconsequential